Wikipedia/Code/WMFArticleListTableViewController.m

```diff
@@ -111,15 +111,21 @@ - (nullable UIViewController *)previewingContext:(id<UIViewControllerPreviewing>
     }
     [[PiwikTracker wmf_configuredInstance] wmf_logActionPreviewInContext:self contentType:contentType];
 
-    if (self.delegate) {
-        return [self.delegate listViewController:self viewControllerForPreviewingArticleURL:url];
-    } else {
-        return [[WMFArticleViewController alloc] initWithArticleURL:url dataStore:self.dataStore];
+    UIViewController *vc = self.delegate ?
+        [self.delegate listViewController:self viewControllerForPreviewingArticleURL:url] : [[WMFArticleViewController alloc] initWithArticleURL:url dataStore:self.dataStore];
+    
+    if ([[vc class] conformsToProtocol:@protocol(WMFArticlePreviewingActionsDelegate)]){
```

Comment:
Is this the right check? Doesn't self conform to `WMFArticlePreviewingActionsDelegate`?

Do we even need to check a class here?